android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/SharedPreferencesRepository.java

```diff
@@ -0,0 +1,53 @@
+package com.appcoins.sdk.billing;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+public class SharedPreferencesRepository {
+
+  private static final String WALLET_ID_KEY = "WALLET_ID";
+  private static final String MAX_BONUS_KEY = "MAX_BONUS";
+  private static final String MAX_BONUS_TTL_SECONDS_KEY = "MAX_BONUS_TTL";
+  private SharedPreferences sharedPreferences;
+
+  public SharedPreferencesRepository(Context context) {
+
+    this.sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+  }
+
+  public String getWalletId() {
+    return sharedPreferences.getString(WALLET_ID_KEY, null);
+  }
+
+  public void setWalletId(String walletId) {
+    sharedPreferences.edit()
+        .putString(WALLET_ID_KEY, walletId)
+        .apply();
+  }
+
+  public int getMaxBonus() {
+    return sharedPreferences.getInt(MAX_BONUS_KEY, 0);
+  }
+
+  public void setMaxBonus(int maxBonus) {
+    if (maxBonus > 0) {
+      sharedPreferences.edit()
+          .putInt(MAX_BONUS_KEY, maxBonus)
+          .apply();
+      sharedPreferences.edit()
+          .putLong(MAX_BONUS_TTL_SECONDS_KEY, System.currentTimeMillis() / 1000)
+          .apply();
+    }
```

Comment:
We won't use the value equal or below 0. Most of the time it will happen when there's no network and that is being dealt in the presenter class, so I don't think it will be necessary.